### PR TITLE
topo: add PathExistsIn helper function

### PR DESCRIPTION
**Pull request by kortschak**
@btracey Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
*ping*

File: topo/topo.go [hunk 1]
```diff
@@ -38,6 +38,16 @@ func IsPathIn(g graph.Graph, path []graph.Node) bool {
...
 }
 
+// PathExistsIn returns whether there is a path in g starting at from extending
+// to to.
+//
+// PathExistsIn exists as a helper function. If many tests for path existence
+// are being performed, other approaches will be more efficient.
+func PathExistsIn(g graph.Graph, from, to graph.Node) bool {
+	var t traverse.BreadthFirst
+	return t.Walk(g, from, func(n graph.Node, _ int) bool { return n.ID() == to.ID() }) != nil
```
Comment (btracey) [inline topo/topo.go]: At some point in my debugging, I thought the following line was necessary. `n.ID() != from.ID() && n.ID() == to.ID()`. Not sure if that's actually true.

Comment (kortschak) [inline topo/topo.go]: I don't think that's true, but if you can find an example that behaves unexpectedly I can discuss/fix.

If we preclude n == from, then you do not get a path from n to n, which is a valid path in any G where n∈V.

Comment (btracey) [inline topo/topo.go]: I'm not sure what you mean by n∈V (n is always in the nodes of the graph), but your general point is well taken.

**Comment by btracey** [comment]
LGTM

File: topo/topo.go (hunk 1 above)
Comment (kortschak) [inline topo/topo.go]: >n is always in the nodes of the graph

Yes, exactly :)

Comment (btracey) [inline topo/topo.go]: Oh, I understand now. Sorry.

**Pull request merged**